sphinx_gallery/gen_rst.py

```diff
@@ -438,46 +464,38 @@ def execute_code_block(src_file, code_block, example_globals,
     orig_stdout = sys.stdout
     src_file = block_vars['src_file']
 
-    try:
-        # First cd in the original example dir, so that any file
-        # created by the example get created in this directory
-        os.chdir(os.path.dirname(src_file))
-        sys.stdout = my_stdout = MixedEncodingStringIO()
+    # First cd in the original example dir, so that any file
+    # created by the example get created in this directory
+    my_stdout = MixedEncodingStringIO()
+    os.chdir(os.path.dirname(src_file))
+    sys.stdout = my_stdout
 
+    try:
+        code_ast = ast.parse(code_block, src_file)
+        ast.increment_lineno(code_ast, lineno - 1)
         t_start = time()
         # don't use unicode_literals at the top of this file or you get
         # nasty errors here on Py2.7
-        exec(code_block, example_globals)
+        exec(compile(code_ast, src_file, 'exec'), example_globals)
         time_elapsed = time() - t_start
 
+    except Exception:
+        sys.stdout = orig_stdout
+        images_rst = handle_exception(sys.exc_info(), src_file, block_vars,
```

Comment:
This is a wrong logic/naming of sphinx-gallery. I've been trying to get rid of this in PR #239, the traceback is not an image_rst, but a formatted output. Since later on we format stdout for the successful output, I believe `handle_exceptions` should save to that variable too to keep consistency.